src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -4923,5 +4923,89 @@ static void F3()
                 //             F1();
                 Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureVariable, "F1()").WithArguments("F1").WithLocation(11, 13));
         }
+
+        [Fact, WorkItem(38240, "https://github.com/dotnet/roslyn/issues/38240")]
+        public void StaticLocalFunctionLocalFunctionDelegateReference()
+        {
+            var source =
+@"#pragma warning disable 8321
+using System;
+class C
+{
+    static void M()
+    {
+        void F1() {}
+
+        static void F2()
+        {
+            Action a = F1;
+        }
```

Comment:
>} [](start = 8, length = 1)

Please also test explicit delegate creation for these cases:
```
_ = new Action(F1);
```

Reply:
@cston Should I do that for all of the tested situations?

Reply:
Yes, please test explicit delegate creation for each of the cases below that currently test MethodGroup conversions. Thanks.

---
In reply to: [317275866](https://github.com/dotnet/roslyn/pull/38251#discussion_r317275866) [](ancestors = 317275866)

Reply:
@cston Good catch! That did reveal some more situations that will need to be fixed, including [those that will require another breaking change, I think](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKHQAgGFcBvXfC/Ae2ACs4BjGfAMwG5zLOKUAWfALIAKAJTd8ZHJWnU6jZgBsOUmRXHSUGAGz4++AGIZR6ypNWqAbhASsArgDsGGfAF589uAHddAVgA8AJb2MAB8QiwAdADicDAAEtAAFoRUACZwIsrmMiwOTqJZ2ZRWNrmOaK7uXr6BwWEK0bEJUMlpGYVFduUFJhQAvuIDOH1AA=).

I'll work on providing more details and a fix tomorrow and then update this PR.

Reply:
@cston I have now added tests for explicit delegate creation (including some new ones), adjusted the code so that they pass and added note about a breaking change required for that.